freqtrade/freqtradebot.py

```diff
@@ -1708,30 +1711,66 @@ def replace_order(self, order: dict, order_obj: Optional[Order], trade: Trade) -
+        :param trade: Trade object of the trade we're analyzing
+        :param reason: The reason for that cancellation
+        :param sides: The sides where cancellation should take place
+        :return: None
+        """
+
+        for open_order in trade.open_orders:
+            try:
+                order = self.exchange.fetch_order(open_order.order_id, trade.pair)
+            except ExchangeError:
+                logger.info("Can't query order for %s due to %s", trade, traceback.format_exc())
+                continue
+
+            if order["side"] in sides:
+                if order["side"] == trade.entry_side:
+                    self.handle_cancel_enter(trade, order, open_order, reason, replacing)
+
+                elif order["side"] == trade.exit_side:
+                    self.handle_cancel_exit(trade, order, open_order, reason)
+
     def cancel_all_open_orders(self) -> None:
         """
         Cancel all orders that are currently open
         :return: None
         """
 
         for trade in Trade.get_open_trades():
-            for open_order in trade.open_orders:
-                try:
-                    order = self.exchange.fetch_order(open_order.order_id, trade.pair)
-                except ExchangeError:
-                    logger.info("Can't query order for %s due to %s", trade, traceback.format_exc())
-                    continue
+            self.cancel_open_orders_of_trade(
+                trade, [trade.entry_side, trade.exit_side], constants.CANCEL_REASON["ALL_CANCELLED"]
+            )
 
-                if order["side"] == trade.entry_side:
-                    self.handle_cancel_enter(
-                        trade, order, open_order, constants.CANCEL_REASON["ALL_CANCELLED"]
-                    )
+        Trade.commit()
 
-                elif order["side"] == trade.exit_side:
-                    self.handle_cancel_exit(
-                        trade, order, open_order, constants.CANCEL_REASON["ALL_CANCELLED"]
+    def handle_similar_open_order(self, trade: Trade, side: str, amount: float) -> bool:
+        """
+        Keep existing open order if same amount and side, otherwise cancel
+        :return: True if an existing similar order was found
+        """
+        if trade.has_open_orders:
+            oo = trade.select_order(side, True)
+            if oo is not None:
+                open_order_side = oo.side
+                open_order_amount = oo.amount
+
+                if (side == open_order_side) & (amount == open_order_amount):
+                    return True
+                else:
+                    # cancel open order of this trade if order is different
+                    self.cancel_open_orders_of_trade(
+                        trade, [trade.entry_side], constants.CANCEL_REASON["REPLACE"], True
                     )
-        Trade.commit()
+                    Trade.commit()
+                    return False
+
+        return False
```

Comment:
The way this works currently completely breaks the replace-order functionality - where the asumption is that the `amount` remains identical - but the `price` changes.

I think this might need 2 additions - one being price (an order for 100 XPR at a different price should be considered a different order) - and the other (maybe?) being a time component (maybe, to be discussed?).

Reply:
I'm totally agree about the fact that price component should be added to this.

Not convinced yet about the time part being the best way to handle edge cases.
Could you elaborate on what case you want to manage?

Indeed I could see potential order replacement infinite loop issues if we base ourselves on price differences when price is based on fluctuating order-book values.

Other solution could be a limit of order replacement number for each new candle analysis?

Reply:
> Could you elaborate on what case you want to manage?

I don't think it would be wrong to "replace" the order once per candle - as it might incur new pricing (obviously pricing could change more often - but that's where i thought about the time component).

Reply:
Agree, let's do this! 
Do you have in mind other places when we added such limit, I'll implement it in a similar way

Reply:
I'm thinking about cases when the strategy is using a relatively high timeframe like 4h, but the user want to adjust an unfilled entry price maybe after 15 minutes.
In that case, if the adjustment is limited to once by candle, the new order won't be created even with different entry price? User will have to use **`check_entry_timeout`** first?

Reply:
Do you still think the time limit is required even when the price is checked before order replacement?

Reply:
> I'm thinking about cases when the strategy is using a relatively high timeframe like 4h, but the user want to adjust an unfilled entry price maybe after 15 minutes.

well i think we're thinking about a different comparison
i'd compare by price - sure - and if price is (significantly) different - we must replace the order.

The problematic part is the "significantly" ... but what's significant? (that also scales across different price ranges)

because of that - my thought was to "time out" the price logic once per candle - so if you issue a "replace order" after 4h - we replace the order (irrelevant if the price changed or not).

if the price-change detection is solid ... might not be needed - only tests will tell though :laughing:

Reply:
It is clear that anything subjective is complicated to be implemented in a "one fit all" manner.
So, let's test if a fix is needed, then we will see what we can do 👍🏽